packages/studio-ui/src/web/components/Layout/TopNav/DeployCloudBtn.tsx

```diff
@@ -0,0 +1,111 @@
+import { Icon, Tooltip, Spinner } from '@blueprintjs/core'
+import axios from 'axios'
+import { lang, toast } from 'botpress/shared'
+import classNames from 'classnames'
+import { Training } from 'common/nlu-training'
+import moment from 'moment'
+import React, { useCallback, useEffect, useState } from 'react'
+import { connect } from 'react-redux'
+import { RootReducer } from '~/reducers'
+
+import style from './style.scss'
+
+interface Props {
+  trainSession: Training
+}
+
+const NeedTraining = () => {
+  return <div className={style.tooltip}>{lang.tr('topNav.deploy.tooltip.needsTraining')}</div>
+}
+const DeployInfo = ({ lastImportDate }) => {
+  const importDateStr = useCallback(() => {
+    if (!lastImportDate) {
+      return null
+    }
+
+    return moment(lastImportDate).fromNow()
+  }, [lastImportDate])
+
+  return (
+    <div className={style.tooltip}>
+      {lastImportDate
+        ? [lang.tr('topNav.deploy.tooltip.lastDeploy'), importDateStr()].join(' ')
+        : lang.tr('topNav.deploy.tooltip.deployBot')}
+    </div>
+  )
+}
+const Deploying = () => {
+  return <div className={style.tooltip}>{lang.tr('topNav.deploy.tooltip.deploying')}</div>
+}
+
+const DeployCloudBtn = (props: Props) => {
+  const { trainSession } = props
+  const [lastImportDate, setLastImportDate] = useState(null)
+  const [deployLoading, setDeployLoading] = useState(false)
+
+  useEffect(() => {
+    axios
+      .get(`${window.STUDIO_API_PATH}/cloud/info`)
+      .then((res) => res.data)
+      .then((data) => {
+        const { imported_on } = data
+        setLastImportDate(imported_on)
+      })
+      .catch((e) => {})
```

Comment:
I added try/catch where I could instead of promises but I could not add a try/catch there as that func needs to be awaited, and you can't add an await function to `useEffect`.